runelite-client/src/main/java/net/runelite/client/plugins/chatcommands/SkillAbbreviations.java

```diff
@@ -85,6 +63,41 @@
 	 */
 	static String getFullName(String abbrev)
 	{
-		return MAP.getOrDefault(abbrev.toUpperCase(), abbrev);
+		if (ALT_NAMES.containsKey(abbrev.toUpperCase()))
+		{
+			return ALT_NAMES.getOrDefault(abbrev.toUpperCase(), abbrev);
+		}
+		else
+		{
+			return getFullNameStartingWith(abbrev);
+		}
+	}
+
+	/**
+	 * Takes the first few letters of a skill, and if exactly one skill starts
+	 * with them, expands it into its full canonical name. Case-insensitive.
+	 *
+	 * @param abbrev First few letters of a skill.
+	 * @return Full skill name if recognized, else the original string.
+	 */
+	private static String getFullNameStartingWith(String abbrev)
+	{
+		List<Skill> matches = new ArrayList<>();
+		for (Skill skill : Skill.class.getEnumConstants())
+		{
+			if (skill.getName().toUpperCase().startsWith(abbrev.toUpperCase()))
+			{
+				matches.add(skill);
```

Comment:
probably just do early return here with first match and then at the end just return the original input? Not sure, I guess this logic is here to check for ambiguous matches, but first match is as good imo.

Reply:
Forgot to comment back on this, but the loop's been cleaned up now.